### PR TITLE
feat(modbus-cli): add option to print out different encoding options

### PR DESCRIPTION
**Pull request by tretmar**
With the option "all", you can now print out different encodings for register results - similar to https://www.scadacore.com/tools/programming-calculators/online-hex-converter/

### CODE REVIEW

File: cmd/modbus-cli/main.go [hunk 1]
```diff
@@ -207,6 +211,108 @@ func resultToRawString(r []byte, startReg int) (string, error) {
...
+		}
+		midLittleFloat32, err := resultToString(result, binary.LittleEndian, "float32")
+		if err != nil {
+			return "", err
+		}
+
+		return strings.Join([]string{
+			fmt.Sprintf("INT32  - Big Endian (ABCD):    %s", bigInt32),
+			fmt.Sprintf("INT32  - Little Endian (DCBA): %s", littleInt32),
+			fmt.Sprintf("INT32  - Mid-Big Endian (BADC):    %s", midBigInt32),
```
Comment (frzifus) [inline cmd/modbus-cli/main.go:297]: why is the distance different? Could we use something like `\t` to keep it in a row?

**Review by frzifus — COMMENTED** [review]
nice idea :smiley:

File: cmd/modbus-cli/main.go (hunk 1 above)
Comment (andig) [reply cmd/modbus-cli/main.go:297]: Use a `text/tabwriter`?

**Review by frzifus — COMMENTED** [review]
I left a few small comments. And have another baisc one, do we still need `  -  ` as a distancer? wdyt?

File: cmd/modbus-cli/main.go [hunk 2]
```diff
@@ -207,6 +212,109 @@ func resultToRawString(r []byte, startReg int) (string, error) {
...
+		littleInt16, err := resultToString(result, binary.LittleEndian, "int16")
+		if err != nil {
+			return "", err
+		}
+
+		return strings.Join([]string{
+			fmt.Sprintf("INT16  - Big Endian (AB):    %s", bigInt16),
+			fmt.Sprintf("INT16  - Little Endian (BA): %s", littleInt16),
+			fmt.Sprintf("UINT16 - Big Endian (AB):    %s", bigUint16),
+			fmt.Sprintf("UINT16 - Little Endian (BA): %s", littleUint16),
```
[the remark this answers is not in the document]
Comment (tretmar) [reply cmd/modbus-cli/main.go]: yeah, I totally forgot to use it for the 16 bit types as well :see_no_evil:

File: cmd/modbus-cli/main.go [hunk 3]
```diff
@@ -207,6 +212,109 @@ func resultToRawString(r []byte, startReg int) (string, error) {
...
+		fmt.Fprintf(w, "INT32\t-\tBig Endian (ABCD):\t%s\t\n", bigInt32)
+		fmt.Fprintf(w, "INT32\t-\tLittle Endian (DCBA):\t%s\t\n", littleInt32)
+		fmt.Fprintf(w, "INT32\t-\tMid-Big Endian (BADC):\t%s\t\n", midBigInt32)
+		fmt.Fprintf(w, "INT32\t-\tMid-Little Endian (CDAB):\t%s\t\n", midLittleInt32)
+		fmt.Fprintln(w, "\t")
+		fmt.Fprintf(w, "UINT32\t-\tBig Endian (ABCD):\t%s\t\n", bigUint32)
+		fmt.Fprintf(w, "UINT32\t-\tLittle Endian (DCBA):\t%s\t\n", littleUint32)
+		fmt.Fprintf(w, "UINT32\t-\tMid-Big Endian (BADC):\t%s\t\n", midBigUint32)
+		fmt.Fprintf(w, "UINT32\t-\tMid-Little Endian (CDAB):\t%s\t\n", midLittleUint32)
+		fmt.Fprintln(w, "\t")
```
[the remark this answers is not in the document]
Comment (tretmar) [reply cmd/modbus-cli/main.go:313]: When you do this, your output is not alligned beautifully anymore. I prefer the one where everything is alligned, wdyt?

Without:
```
INT32  Big Endian (ABCD):         -2129464508  
INT32  Little Endian (DCBA):      1156780673   
INT32  Mid-Big Endian (BADC):     310461683    
INT32  Mid-Little Endian (CDAB):  -213614318   

UINT32  Big Endian (ABCD):         2165502788  
UINT32  Little Endian (DCBA):      1156780673  
UINT32  Mid-Big Endian (BADC):     310461683   
UINT32  Mid-Little Endian (CDAB):  4081352978  

Float32  Big Endian (ABCD):         -0.000000                                 
Float32  Little Endian (DCBA):      1944.578247                               
Float32  Mid-Big Endian (BADC):     0.000000                                  
Float32  Mid-Little Endian (CDAB):  -15568665179727516745525255733248.000000 
```

With /t:

```
INT32    Big Endian (ABCD):         -2129464508  
INT32    Little Endian (DCBA):      1156780673   
INT32    Mid-Big Endian (BADC):     310461683    
INT32    Mid-Little Endian (CDAB):  -213614318   
         
UINT32   Big Endian (ABCD):         2165502788  
UINT32   Little Endian (DCBA):      1156780673  
UINT32   Mid-Big Endian (BADC):     310461683   
UINT32   Mid-Little Endian (CDAB):  4081352978  
         
Float32  Big Endian (ABCD):         -0.000000                                 
Float32  Little Endian (DCBA):      1944.578247                               
Float32  Mid-Big Endian (BADC):     0.000000                                  
Float32  Mid-Little Endian (CDAB):  -15568665179727516745525255733248.000000  
```

File: cmd/modbus-cli/main.go [hunk 4]
```diff
@@ -207,6 +212,109 @@ func resultToRawString(r []byte, startReg int) (string, error) {
...
+			return "", err
+		}
+
+		buf := new(bytes.Buffer)
+		w := tabwriter.NewWriter(buf, 0, 0, 2, ' ', 0)
+		fmt.Fprintf(w, "INT32\t-\tBig Endian (ABCD):\t%s\t\n", bigInt32)
+		fmt.Fprintf(w, "INT32\t-\tLittle Endian (DCBA):\t%s\t\n", littleInt32)
+		fmt.Fprintf(w, "INT32\t-\tMid-Big Endian (BADC):\t%s\t\n", midBigInt32)
+		fmt.Fprintf(w, "INT32\t-\tMid-Little Endian (CDAB):\t%s\t\n", midLittleInt32)
+		fmt.Fprintln(w, "\t")
```
[the remark this answers is not in the document]
Comment (tretmar) [reply cmd/modbus-cli/main.go:308]: same as before

File: cmd/modbus-cli/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (frzifus) [reply cmd/modbus-cli/main.go:313]: Ah its not aligned across all sections. For me, both would be fine, Ive no strong preference. :no_good:

**Review by frzifus — APPROVED** [review]
lgtm, up to the others :)

**Review by Rania-Zoghlami — COMMENTED** [review]
LGTM :+1:

**Review by guelfey — APPROVED** [review]

**Pull request merged**